ppc64le: nightly build failure on GCC<=6

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

Please:

* Read the documentation to test with the latest developer build.
* Check if other person has already created the same issue to avoid duplicates. You can comment on it if there already is an issue.
* Try to be as detailed as possible in your report.
* Report only one problem per created issue.


This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => after #13334
- Operating System / Platform => ppc64le GNU/Linux
- Compiler => gcc4.9, gcc5, gcc6

##### Detailed description

````
In file included from /worker/buildbot/Power8_Linux_gcc-5__opencv/opencv/modules/core/include/opencv2/core/hal/intrin.hpp:158:0,
                 from /worker/buildbot/Power8_Linux_gcc-5__opencv/opencv/modules/core/src/precomp.hpp:85,
                 from /worker/buildbot/Power8_Linux_gcc-5__opencv/build/modules/core/opencv_core_pch_dephelp.cxx:1:
/worker/buildbot/Power8_Linux_gcc-5__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp: In function 'double cv::hal_baseline::v_reduce_sum(const cv::hal_baseline::v_float64x2&)':
/worker/buildbot/Power8_Linux_gcc-5__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:721:62: error: invalid parameter combination for AltiVec intrinsic
     return vec_extract(vec_add(a.val, vec_sld(a.val, a.val, 8)), 0);
                                                              ^
/worker/buildbot/Power8_Linux_gcc-5__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:721:67: error: invalid parameter combination for AltiVec intrinsic
     return vec_extract(vec_add(a.val, vec_sld(a.val, a.val, 8)), 0);
                                                                   ^
````
implementation of v_reduce_sum(const cv::hal_baseline::v_float64x2&), vec_sld doesn't support double precision

CI: https://ocv-power.imavr.com/#/builders/9/builds/574
 
##### Steps to reproduce

build OpenCV on ppc64le

--- a/modules/core/include/opencv2/core/hal/intrin_vsx.hpp
+++ b/modules/core/include/opencv2/core/hal/intrin_vsx.hpp
@@ -718,7 +718,7 @@ OPENCV_HAL_IMPL_VSX_REDUCE_OP_4(v_float32x4, vec_float4, float, min, vec_min)
 
 inline double v_reduce_sum(const v_float64x2& a)
 {
-    return vec_extract(vec_add(a.val, vec_sld(a.val, a.val, 8)), 0);
+    return vec_extract(vec_add(a.val, vec_permi(a.val, a.val, 3)), 0);
 }
 
 #define OPENCV_HAL_IMPL_VSX_REDUCE_OP_8(_Tpvec, _Tpvec2, scalartype, suffix, func) \

